rise_set/visibility.py

```diff
@@ -233,6 +233,43 @@ def get_zenith_distance_intervals(self, target, target_intervals, chunksize=date
         """
         return self._get_chunked_intervals(target, target_intervals, self._add_zenith_interval, self.zenith_blind_spot, chunksize)
 
+    def get_moon_phase_intervals(self, target_intervals, max_moon_phase=1.0, chunksize=datetime.timedelta(minutes=30)):
+        """ Returns the intervals in which the target is visible and the moon phase does not exceed max_moon_phase
+
+        Args:
+            target_intervals (list): A list of datetime tuples that represent the above horizon intervals for the target.
+                                     Returned by get_target_intervals()
+            max_moon_phase (float): max allowablable fractional moon phase from 0 to 1 (full moon).
+            chunksize (timedelta): The time delta over which to calculate if the target intervals are out of range of
+                                   the zenith.
+        Returns:
+            list: A list of tuples of start/end datetime pairs that make up the intervals over which this target is visible and the moon phase is less than max_moon_phase
+        """
+        # If target intervals are empty, no point in calculating moon phases
+        if not target_intervals:
+            return target_intervals
+        bad_intervals = []
+        moon_up_intervals = self.get_target_intervals(target="moon", up=True)
+        target_moon_up_intervals = intersect_intervals(target_intervals, moon_up_intervals)
+        # Here we collect the bad intervals where the moon is up and the moon phase is greater than the max constraint
+        for start, end in target_moon_up_intervals:
+            chunkstart = start
```

Comment:
nitpick but chunk_start, chunk_end would read a bit better I think. pep8 recommends separating words with underscores to enhance readability.